src/Symfony/Component/Messenger/Retry/MultiplierRetryStrategy.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Retry;
+
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Stamp\RetryCountStamp;
+
+/**
+ * A retry strategy with a constant or exponential retry delay.
+ *
+ * For example, if $delayMilliseconds=10000 & $multiplier=1 (default),
+ * each retry will wait exactly 10 seconds.
+ *
+ * But if $delayMilliseconds=10000 & $multiplier=2:
+ *      * Retry 1: 10 second delay
+ *      * Retry 2: 20 second delay (10000 * 2 = 20000)
+ *      * Retry 3: 40 second delay (20000 * 2 = 40000)
+ *
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ *
+ * @experimental in 4.3
+ * @final
+ */
+class MultiplierRetryStrategy implements RetryStrategyInterface
+{
+    private $maxRetries;
+
+    private $delayMilliseconds;
+
+    private $multiplier;
+
+    private $maxDelayMilliseconds;
+
+    /**
+     * @param int $maxRetries           The maximum number of time to retry (0 means indefinitely)
+     * @param int $delayMilliseconds    Amount of time to delay (or the initial value when multiplier is used)
+     * @param int $multiplier           Multiplier to apply to the delay each time a retry occurs
+     * @param int $maxDelayMilliseconds Maximum delay to allow (0 means no maximum)
+     */
+    public function __construct(int $maxRetries = 3, int $delayMilliseconds = 1000, int $multiplier = 1, int $maxDelayMilliseconds = 0)
+    {
+        $this->maxRetries = $maxRetries;
+
+        if ($delayMilliseconds < 0) {
+            throw new \InvalidArgumentException(sprintf('Delay must be greater than or equal to zero: "%s" passed', $delayMilliseconds));
```

Comment:
Missing dots at the end of the exception messages (same for other exception messages).